commit 51ec138c6416e9ed2ba0eae3af5f0ea8a90ae44b
tree 4c451864585014378922e16761be0a5225455166
parent 0e81c666dbf95546b3d9ea6ff7d29ea19b988950
author Al Viro <viro@ftp.linux.org.uk> 1184529591 +0100
committer Linus Torvalds <torvalds@woody.linux-foundation.org> 1184542851 -0700

    ieee1394: forgotten dereference...
    
    Going through the string and waiting for _pointer_ to become '\0'
    is not what the authors meant...
    
    Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>
    Acked-by: Ben Collins <ben.collins@ubuntu.com>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

diff --git a/drivers/ieee1394/nodemgr.c b/drivers/ieee1394/nodemgr.c
index c4d3d41..51a1206 100644
--- a/drivers/ieee1394/nodemgr.c
+++ b/drivers/ieee1394/nodemgr.c
@@ -283,7 +283,7 @@ static ssize_t fw_show_##class##_##td_kv
 	memcpy(buf,							\
 	       CSR1212_TEXTUAL_DESCRIPTOR_LEAF_DATA(class->td_kv),	\
 	       len);							\
-	while ((buf + len - 1) == '\0')					\
+	while (buf[len - 1] == '\0')					\
 		len--;							\
 	buf[len++] = '\n';						\
 	buf[len] = '\0';						\
